test/account-creation.js

```diff
@@ -59,45 +57,36 @@ describe('Identity Provider', function () {
     })
   })
 
-  describe('generating a certificate', function () {
+  describe('generating a certificate', () => {
     beforeEach(function () {
       rm('accounts/nicola.localhost')
     })
     after(function () {
       rm('accounts/nicola.localhost')
     })
 
-    it('should generate a certificate if spkac is valid', function (done) {
+    it('should generate a certificate if spkac is valid', (done) => {
       var spkac = read('example_spkac.cnf')
       var subdomain = supertest.agent('https://nicola.' + host)
       subdomain.post('/api/accounts/new')
-        .send('username=nicola')
-        .expect(200)
-        .end(function (err, req) {
-          if (err) return done(err)
-
-          subdomain.post('/api/accounts/cert')
-            .send('spkac=' + spkac + '&webid=https%3A%2F%2Fnicola.localhost%3A3457%2Fprofile%2Fcard%23me')
-            .expect('Content-Type', /application\/x-x509-user-cert/)
-            .expect(200)
-            .end(done)
-        })
-    })
+        .send('username=nicola&spkac=' + spkac)
+        .expect('Content-Type', /application\/x-x509-user-cert/)
+        .expect(200, done)
+    }).timeout(20000)
```

Comment:
I wholly agree :) These tests were timing out for me, before I started the refactor (they were running about 10-12 seconds, on my laptop), which is way too long, but that's part of the reason I'm trying to replace this authentication method.